src/core/SpriteAnimator.tsx

```diff
@@ -1,98 +1,177 @@
+/* eslint react-hooks/exhaustive-deps: 1 */
 import * as React from 'react'
-import { useFrame, Vector3, MeshProps } from '@react-three/fiber'
+import { useFrame, Vector3 } from '@react-three/fiber'
 import * as THREE from 'three'
 import { Instances, Instance } from './Instances'
+import { Billboard } from './Billboard'
 import { useSpriteLoader } from './useSpriteLoader'
 
-type CommonMeshProps = {
-  frustumCulled?: boolean
-  renderOrder?: number
-  visible?: boolean
-  userData?: any
-  matrixAutoUpdate?: boolean
+// Frame-related types
+type Size = {
+  w: number
+  h: number
 }
 
+type FrameData = {
+  frame: {
+    x: number
+    y: number
+  }
+  sourceSize: Size
+}
+
+type Frames = Record<string, FrameData[]> | FrameData[]
```

Comment:
I think the 2nd form `FrameData[]` does not bring anything, but brings complexity into the code to deal with this alternate format

i think you should remove it

Reply:
The JSON format that this component supports could be either an array (JSON Array) or an object (JSON Hash), this is how some software export them (TexturePacker for example) so we need to support  both options, the code deals with this so the end-user doesn't have to pick between the two

Reply:
makes sense then ;)